src/Tools/AnalyzerRunner/DiagnosticAnalyzerRunner.cs

```diff
@@ -0,0 +1,451 @@
+            {
+                WriteTelemetry(pair.Key.GetType().Name, pair.Value);
+            }
+
+            WriteLine($"Execution times (ms):", ConsoleColor.DarkCyan);
+            var longestAnalyzerName = telemetryInfoDictionary.Select(x => x.Key.GetType().Name.Length).Max();
+            foreach (var pair in telemetryInfoDictionary.OrderBy(x => x.Key.GetType().Name, StringComparer.OrdinalIgnoreCase))
+            {
+                WriteExecutionTimes(pair.Key.GetType().Name, longestAnalyzerName, pair.Value);
+            }
+        }
+
+        private static void WriteTelemetry(string analyzerName, AnalyzerTelemetryInfo telemetry)
+        {
+            WriteLine($"Statistics for {analyzerName}:", ConsoleColor.DarkCyan);
+            WriteLine($"Concurrent:                     {telemetry.Concurrent}", telemetry.Concurrent ? ConsoleColor.White : ConsoleColor.DarkRed);
+            WriteLine($"Execution time (ms):            {telemetry.ExecutionTime.TotalMilliseconds}", ConsoleColor.White);
+
+            WriteLine($"Code Block Actions:             {telemetry.CodeBlockActionsCount}", ConsoleColor.White);
+            WriteLine($"Code Block Start Actions:       {telemetry.CodeBlockStartActionsCount}", ConsoleColor.White);
+            WriteLine($"Code Block End Actions:         {telemetry.CodeBlockEndActionsCount}", ConsoleColor.White);
+
+            WriteLine($"Compilation Actions:            {telemetry.CompilationActionsCount}", ConsoleColor.White);
+            WriteLine($"Compilation Start Actions:      {telemetry.CompilationStartActionsCount}", ConsoleColor.White);
+            WriteLine($"Compilation End Actions:        {telemetry.CompilationEndActionsCount}", ConsoleColor.White);
+
+            WriteLine($"Operation Actions:              {telemetry.OperationActionsCount}", ConsoleColor.White);
+            WriteLine($"Operation Block Actions:        {telemetry.OperationBlockActionsCount}", ConsoleColor.White);
+            WriteLine($"Operation Block Start Actions:  {telemetry.OperationBlockStartActionsCount}", ConsoleColor.White);
+            WriteLine($"Operation Block End Actions:    {telemetry.OperationBlockEndActionsCount}", ConsoleColor.White);
+
+            WriteLine($"Semantic Model Actions:         {telemetry.SemanticModelActionsCount}", ConsoleColor.White);
+            WriteLine($"Symbol Actions:                 {telemetry.SymbolActionsCount}", ConsoleColor.White);
+            WriteLine($"Symbol Start Actions:           {telemetry.SymbolStartActionsCount}", ConsoleColor.White);
+            WriteLine($"Symbol End Actions:             {telemetry.SymbolEndActionsCount}", ConsoleColor.White);
+            WriteLine($"Syntax Node Actions:            {telemetry.SyntaxNodeActionsCount}", ConsoleColor.White);
+            WriteLine($"Syntax Tree Actions:            {telemetry.SyntaxTreeActionsCount}", ConsoleColor.White);
```

Comment:
Maybe we can also add a metric for the number of diagnostics reported on each context - where applicable. (i.e. figure out busy/wasted work)

Reply:
This is all just moved from Program.cs